lib/_tls_wrap.js

```diff
@@ -247,7 +247,19 @@ function TLSSocket(socket, options) {
   // starting the flow of the data
   this.readable = true;
   this.writable = true;
-  this.read(0);
+
+  // Don't read until user-land had a chance to setup listeners
+  var self = this;
+  process.nextTick(function() {
+    // Socket already has some buffered data - emulate receiving it
+    if (socket && socket._readableState && socket._readableState.length) {
+      var buf;
+      while ((buf = socket.read()) !== null)
+        self._handle.receive(buf);
+    }
+
+    self.read(0);
+  });
```

Comment:
It should be fine to make use of `nextTick` args in this PR. It's likely that the Node patches will be merged on top of iojs when the convergence WG starts up (so we're not worried about porting back patches for now.)